Blogpost on multilingual documentation

eliocamp opened the pull request:
This adds

eliocamp commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,199 @@
+Package authors face the dilemma of either making their package inaccessible to their target demographic or isolating it from the wider R ecosystem.
+
+The developer of the [utilsIPEA](https://cran.r-project.org/web/packages/utilsIPEA/index.html) package publicly expressed [the need for bilingual documentation](https://stackoverflow.com/questions/37288823/bilingual-english-and-portuguese-documentation-in-an-r-package), recognising that his package would be used both by people in Brazil, who might prefer documentation in Portuguese, and the broad ...
+
```
Maybe add something like: 

```suggestion
This is also a recurring question in the rOpenSci Slack. 
```

eliocamp commented:
I added your suggestions and corrected some typos and spelling mistakes.

eliocamp commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -139,7 +139,7 @@ Package authors face the dilemma of either making their package inaccessible to
 The developer of the [utilsIPEA](https://cran.r-project.org/web/packages/utilsIPEA/index.html) package publicly expressed [the need for bilingual documentation](https://stackoverflow.com/questions/37288823/bilingual-english-and-portuguese-documentation-in-an-r-package), recognising that his package would be used both by people in Brazil, who might prefer documentation in Portuguese, and the broad ...
 This problem comes up periodically in the rOpenSci Slack too.
-
+This is also a recurring question in the rOpenSci Slack. 
```
Oops. This is now repeated xD. 
```suggestion

```

yabellini commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,199 @@
+Amazing!
+
+## Come help us!
+
```
Awesome!

yabellini commented:
@eliocamp, thanks for the additions.  😄  
Can you copy the author checklist as a comment and fill it in? https://blogguide.ropensci.org/authortechnical.html#other-posts

@steffilazerte this is ready for your review.

eliocamp commented:
* [ ] I have read the Content Guidelines.
* [ ] I have read the Technical Guidelines.
* [ ] I used or followed the R Markdown or Markdown template.
* [ ] I have followed the Style Guide.
* [ ] I created or updated my author metadata with correct folder name.
* [ ] I have added relevant tags after browsing existing tags (including "community" tag).
* [ ] I have added the "tech notes" tag if this is a technote.
* [ ] I ran `roblog::ro_lint_md()` on index.md (optional).
* [ ] I ran `roblog::ro_check_urls()` on index.md (optional).
* [ ] I ran a spell-check on index.md.

yabellini commented:
@eliocamp, regarding the missing description in the repo, I was talking about the package description in the same place you added the link to a webpage or the documentation,  the release, the Code of Conduct, etc. (upper right corner of the screen).

eliocamp commented:
Ah, yes, you're right!

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,207 @@
+}
+```
+
+We are happy to announce the R Consortium Multilingual Working Group.
````
I would also suggest getting the project listed on the [Working groups](https://www.r-consortium.org/all-projects/isc-working-groups) site as well if you know who to bug about that 😁 
```suggestion
We are happy to announce the [R Consortium Multilingual Working Group](https://github.com/RConsortium/multilingual-documentation-wg).
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,207 @@
+```
+
+We are happy to announce the R Consortium Multilingual Working Group.
+This Working Group will oversee the implementation of multilingual documentation support in R and organise community translation efforts.
````
Did you want to add a sentence or two about how/when this group formed? A little bit of background?

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,207 @@
+
+We are happy to announce the R Consortium Multilingual Working Group.
+This Working Group will oversee the implementation of multilingual documentation support in R and organise community translation efforts.
+Our first project is the (experimental) **rhelpi18n** package, which adds multilingual documentation support!
```
```suggestion
Our first project is the (experimental) [**rhelpi18n**](https://github.com/eliocamp/rhelpi18n) package, which adds multilingual documentation support!
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,207 @@
+
+```{r, eval=FALSE}
+# install.packages("pak")
+pak::pak(c("eliocamp/rhelpi18n", "eliocamp/base.es"))
````
I tried to install these packages like this, and for some reason base.es could only be installed *after* rhelpi18n. Perhaps there's an unstated dependency issue? 

I was able to install it by running
```suggestion
pak::pak("eliocamp/rhelpi18n")
pak::pak("eliocamp/base.es")
```

eliocamp replied:
Indeed! Thanks for catching that! I updated base.es DESCRIPTION, so now it should work correctly.

steffilazerte reviewed (commented):
Hi @eliocamp! 

I have a couple of review comments here, but mainly, I was wondering if you could re-render the index.Rmd file? I think it's slightly out of date, and unfortunately I can't render it myself. I ran into problems trying to run the code as my installation doesn't have .md file for the glue documentation... perhaps my system (Ubuntu) or the way I install packages? Or perhaps I referenced the wrong package location. 

Anyway, if you could render the file for me (feel free to address my comments now or leave them for later), that'd be great! I'm not so concerned that the code works for everyone, as you're very clear that it's experimental 😁 .

I'm looking forward to diving into the rest of this article, thanks!

eliocamp commented:
The problem might be that I'm using the glue documentation from an external folder that *I* have on `/home/user1/Documents/r-packages/glue`. I though about bundling that in the roweb3 repo, but it's quite big and a lot of files. I could add code to download a copy of glue to a temporary folder and use that, if you'd like.

eliocamp commented:
I added some code to download glue to a temp folder so the post is easier to knit. Fixed some typos and added a few words of context for the WG. Let me know what you think.

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+This is a regular package with special fields in the DESCRIPTION specifying which package and version is being translated and to which language.
+Currently, translated strings are implemented as a list with the original string and the translation of each section in the documentation, but we are also looking at implementing this with traditional .po files, which is the mechanism used by R to translate messages and errors.
+
+The rhelpi18n package, on the other hand, modifies `utils:::.getHelpFile()`, which all help functions use to fetch documentation strings.
```
```suggestion
The rhelpi18n package modifies `utils:::.getHelpFile()`, which all help functions use to fetch documentation strings.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+Currently, translated strings are implemented as a list with the original string and the translation of each section in the documentation, but we are also looking at implementing this with traditional .po files, which is the mechanism used by R to translate messages and errors.
+
+The rhelpi18n package, on the other hand, modifies `utils:::.getHelpFile()`, which all help functions use to fetch documentation strings.
+It adds a few lines to check for relevant translation modules and replaces the original strings with the translated strings on the fly.
```
```suggestion
It adds a few lines of code which check for relevant translation modules (such as base.es) and replaces the original strings with the translated strings on the fly.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+The rhelpi18n package, on the other hand, modifies `utils:::.getHelpFile()`, which all help functions use to fetch documentation strings.
+It adds a few lines to check for relevant translation modules and replaces the original strings with the translated strings on the fly.
+
+This package also provides tools for developers to create new translation modules.
```
```suggestion
In addition to presenting translated documentation, rhelpi18n also provides tools for developers to create new translation modules.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+
+This package also provides tools for developers to create new translation modules.
+
+```{r, echo=FALSE, message=FALSE, warning=FALSE}
+glue_zip <- file.path(tempdir(), "glue.zip")
+glue_dir <- file.path(tempdir(), "glue-9f88e2179fc7aa6ff11214271fd61bba169f925c")
+dir.create(glue_dir, showWarnings = FALSE, recursive = TRUE)
+glue_url <- "https://github.com/tidyverse/glue/archive/9f88e2179fc7aa6ff11214271fd61bba169f925c.zip"
+
+download.file(glue_url, glue_zip)
+unzip(glue_zip, exdir = tempdir())
````
Ah! That was the problem, I thought it was the installed copy of glue, not the source 😁

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+                              module_path = dir)
+```
+
+This creates a translation module with all the functionality and strings that need translating.
````
Consider moving to above the code?
```suggestion
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+download.file(glue_url, glue_zip)
+unzip(glue_zip, exdir = tempdir())
+```
+
````
Moved and edited from below
```suggestion
For example, the following code creates a basic translation module for the glue package, including all the functionality and strings that need translating.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+```{r}
+dir <- file.path(tempdir(), "glue.es")
+rhelpi18n::i18n_module_create(module_name = "glue.es", 
+                              package_path = glue_dir,  # I have a copy of glue here. 
````
```suggestion
                              package_path = glue_dir,  # Path to local copy of glue source code
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+```{r}
+dir <- file.path(tempdir(), "glue.es")
+rhelpi18n::i18n_module_create(module_name = "glue.es", 
+                              package_path = glue_dir,  # I have a copy of glue here. 
````
Not important for the post, but it'd be pretty cool if you could point rhelpi18n to a GitHub repository (or similar)!

eliocamp replied:
Yes. Although that would add a bit of complexity and dependencies. It might also make less transparent which specific version is being translated. But we need to think about this.

steffilazerte replied:
Ah very good point, I guess this is an important consideration whereever the source comes from!

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+Now the heavy work is to go to the "translations" subdirectory and translate all the strings saved (currently) as YAML files.
+For example, the translation file for `glue::as_glue()` is:
+
+```{r,  output.lines=20}
````
I'm not sure you need to share this code?
```suggestion
```{r,  echo = FALSE, output.lines=20}
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+  cat(sep = "\n")
+```
+
+A possible (and crummy) translation would be
````
Crummy sounds a little too disparaging 😁 
```suggestion
A possible (and superficial) translation would be
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
````diff
@@ -0,0 +1,223 @@
+# pak::local_install(dir)
+```
+
+After modifying "as_glue.yaml" and installing the package, now the documentation is in Spanish.
````
```suggestion
After modifying "as_glue.yaml" and installing the package, now the documentation is in Spanish 🎉.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+## Next steps
+
+This is all in the early stages and things are moving fast.
+We are still experimenting with various ways of storing translations and implementations of string replacement.
```
```suggestion
We are still experimenting with various ways to store translations and implement string replacements.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+This is all in the early stages and things are moving fast.
+We are still experimenting with various ways of storing translations and implementations of string replacement.
+It also has several limitations, such as not translating section headings yet.
+We want to add a clear indication that the text is a translation and a convenient way of quickly accessing the original version.
```
```suggestion
We also want to add a clear indication that the text is a translation and a convenient way of quickly accessing the original version.
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+We are still experimenting with various ways of storing translations and implementations of string replacement.
+It also has several limitations, such as not translating section headings yet.
+We want to add a clear indication that the text is a translation and a convenient way of quickly accessing the original version.
+We are also considering adding metadata to strings to flag cases of automatic translations that haven't been reviewed by a human or translations that might be slightly outdated (it would be a shame to not show a translation because someone fixed a small typo in the original text!).
```
I felt like this needed a bit of context, have I understood correctly?
```suggestion
Part of the translations workflow often involves automated translations which are later reviewed by a human. So we are also considering adding metadata to strings to flag cases of automatic translations that haven't been reviewed by a human or translations that might be slightly outdated (it would be a shame to not show a translation because someone fixed a small typo in the original text and it had yet to be reviewed!).
```

eliocamp replied:
Yes!

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+We are also considering adding metadata to strings to flag cases of automatic translations that haven't been reviewed by a human or translations that might be slightly outdated (it would be a shame to not show a translation because someone fixed a small typo in the original text!).
+
+If you want to test the package and report any problems, or you have ideas on how to improve it, you can join the discussion by checking the [open issues](https://github.com/eliocamp/rhelpi18n/issues) of the package and by joining the [Multilingual Documentation Working Group](https://github.com/RConsortium/multilingual-documentation-wg).
+
```
Perhaps all on the same line?
```suggestion
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+tags:
+  - R
+  - packages
+  - multilingual
```
```suggestion
  - multilingual
  - documentation
  - community
```

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+author:
+  - Elio Campitelli
+# Set the date below to the publication date of your post
+date: 2024-06-10
```
When we have a date, change it here and in the folder name too

eliocamp replied:
Great. I'll shoot for the 9th

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
When we have a date, remember to change the folder name too 😁

steffilazerte commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+  - packages
+  - multilingual
+# The summary below will be used by e.g. Mastodon preview cards
+description: "A very short summary of your post (~ 100 characters)"
```
```suggestion
description: "Multilingual documentation coming to an R package near you"
```

steffilazerte reviewed (commented):
Hi @eliocamp!

Wow! This is such exciting stuff, thanks for writing this post for us, I found it really interesting and so cool to see what's in the works!

I'm an opinionated editor, so you'll see that I've made a fair number of suggestions, mostly which are rearranging your sentences and paragraphs into slightly different orders. 

I also suggest adding a bit of a reminder to people in the "why do this?" that translations are awesome and make the world a better place. 

Feel free to accept, reject or modify any of my suggestions, this is your post 😁 Bear in mind that GitHub often hides comments and you need to click on "Show more comments" (this comes up more than you would think!). If you could resolve comments as you're done with them, that'd be great, it helps me make sure nothing was missed. 

The only thing you'll definitely have to change is the date of the post, once we have a date lined up. If you think you'll have time to address the comments by **Monday July 8th**, we can aim to publish on **Tuesday July 9th**, but there's no rush on our part. We have another big post coming out on Thursday, so the next good date to publish would be Tuesday July 16th. 

Thanks again Elio!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [ ] publication date is ok  --> Update once we have some time.
* [X] alternative text of images is informative
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok ("tech notes" for tech notes; "community" for non-staff non-editor)
* [X] YAML field 'preface' is present if necessary

eliocamp commented on content/blog/2024-06-10-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,223 @@
+## Why would you do this?
+
+English is the de facto international language.
+This English hegemony is reflected in R by the use of English in its documentation, such as manual pages, function names, and argument names (why use `mean()` instead of `Mittelwert()`?)
```
I'd prefer not to add an explicit value judgement to this.

steffilazerte replied:
Good point 😁! I re-read the paragraph and realized that I was thinking of it in negative terms. "Hegemony" isn't in and of itself a negative term, but its contemporary usage is generally perceived as negative (i.e. we generally think that hegemonies aren't a good thing).

But you're right that my suggestion would have made the sentence imply that the R ecosystem was "unfortunate", whereas sticking with "The English hegemony" merely leaves the general, global hegemony unfortunate, which most can recognize.

eliocamp commented:
I've updated the date and added some of the suggestions. Let me know what you think. Once it's ready, I'd like to translate it to Spanish.

yabellini commented:
@eliocamp, you can use Babel down to do the translations, and I will review them.  Ping me when it is ready.  
I read the new version, and I think it looks great. There are still two comments active, do you want to resolve them so we know those are solve?

eliocamp commented:
I marked them as resolved. I'll create a deepl account and do the translation today.

eliocamp commented:
I've added the automatic Spanish translation. (BTW, the babeldown package is like magic!)

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+- documentation
+- community
+description: Documentación multilingüe en un paquete R cercano
+social: "La documentación multilingüe llega a un paquete de R cercano @eliocamp@mastodon.social"
```
```suggestion
social: "La documentación multilingüe de paquetes de R al alcance de tu mano @eliocamp@mastodon.social"
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
`````diff
@@ -0,0 +1,217 @@
+}
+````
+
+Nos complace anunciar el nuevo [Grupo de Trabajo Multilingüe del Consorcio R](https://github.com/RConsortium/multilingual-documentation-wg).
`````
```suggestion
Nos complace anunciar el nuevo [Grupo de Trabajo Multilingüe del R Consortium](https://github.com/RConsortium/multilingual-documentation-wg).
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
`````diff
@@ -0,0 +1,217 @@
+````
+
+Nos complace anunciar el nuevo [Grupo de Trabajo Multilingüe del Consorcio R](https://github.com/RConsortium/multilingual-documentation-wg).
+Este Grupo de Trabajo surgió tras los debates mantenidos durante la [Proyecto R Sprint 2023](https://contributor.r-project.org/r-project-sprint-2023/) y supervisará la implementación del soporte de documentación multilingüe en R y organizará los esfuerzos de traducción de la comunidad.
`````
```suggestion
Este Grupo de Trabajo surgió tras los debates mantenidos durante el [R Project Sprint 2023](https://contributor.r-project.org/r-project-sprint-2023/) y supervisará la implementación del soporte de documentación multilingüe en R y organizará los esfuerzos de traducción de la comunidad.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+Este Grupo de Trabajo surgió tras los debates mantenidos durante la [Proyecto R Sprint 2023](https://contributor.r-project.org/r-project-sprint-2023/) y supervisará la implementación del soporte de documentación multilingüe en R y organizará los esfuerzos de traducción de la comunidad.
+Nuestro primer proyecto es el (experimental) [**rhelpi18n**](https://github.com/eliocamp/rhelpi18n) que añade documentación multilingüe.
+
+¿Desea leer la documentación de `base::mean()` (parcialmente) en español?
```
```suggestion
¿Quieres leer la documentación de `base::mean()` (parcialmente) en español?
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+Nuestro primer proyecto es el (experimental) [**rhelpi18n**](https://github.com/eliocamp/rhelpi18n) que añade documentación multilingüe.
+
+¿Desea leer la documentación de `base::mean()` (parcialmente) en español?
+Instale la versión experimental [rhelpi18n](https://github.com/eliocamp/rhelpi18n) y el módulo de traducción de ejemplo base.es:
```
```suggestion
Instala la versión experimental [rhelpi18n](https://github.com/eliocamp/rhelpi18n) y el módulo de traducción de ejemplo base.es:
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+
+Aunque el inglés es la lengua internacional de facto, existen [ventajas](https://ropensci.org/multilingual-publishing/) de incluir recursos en diferentes idiomas.
+Al reducir las barreras lingüísticas, acabamos teniendo una comunidad más grande, fuerte y creativa, con más ideas y recursos.
+Esta hegemonía del inglés se refleja en R por el uso del inglés en su documentación, como las páginas de manual, los nombres de las funciones y los nombres de los argumentos (¿por qué usar `mean()` en lugar de `Mittelwert()`?)
```
```suggestion
Esta hegemonía del inglés se refleja en R por el uso del inglés en su documentación, como las páginas de ayuda, los nombres de las funciones y los nombres de los argumentos (¿por qué usar `mean()` en lugar de `Mittelwert()`?)
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+Esta hegemonía del inglés se refleja en R por el uso del inglés en su documentación, como las páginas de manual, los nombres de las funciones y los nombres de los argumentos (¿por qué usar `mean()` en lugar de `Mittelwert()`?)
+.
+
+Documentación del paquete *puede* puede escribirse en cualquier idioma, y CRAN admite documentación en otros idiomas utilizando el campo "Language" (Idioma) [^1] .
```
```suggestion
La documentación de los paquetes *puede* escribirse en cualquier idioma, y CRAN admite documentación en otros idiomas utilizando el campo "Language" (Idioma) [^1] .
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+.
+
+Documentación del paquete *puede* puede escribirse en cualquier idioma, y CRAN admite documentación en otros idiomas utilizando el campo "Language" (Idioma) [^1] .
+Pero aún así, la gran mayoría de los paquetes están documentados en inglés y el campo [pequeño número de paquetes documentados en otros idi](https://cderv.rbind.io/2018/03/11/non-english-pkg-in-cran/) parecen estar adaptados a su público objetivo.
```
```suggestion
Pero aún así, la gran mayoría de los paquetes están documentados en inglés y el [pequeño número de paquetes documentados en otros idimas](https://cderv.rbind.io/2018/03/11/non-english-pkg-in-cran/) se ajustan aparentemente a su público objetivo.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+
+Documentación del paquete *puede* puede escribirse en cualquier idioma, y CRAN admite documentación en otros idiomas utilizando el campo "Language" (Idioma) [^1] .
+Pero aún así, la gran mayoría de los paquetes están documentados en inglés y el campo [pequeño número de paquetes documentados en otros idi](https://cderv.rbind.io/2018/03/11/non-english-pkg-in-cran/) parecen estar adaptados a su público objetivo.
+Por ejemplo, los paquetes [labstatR](https://cran.r-project.org/web/packages/labstatR/index.html) sirve de complemento al libro italiano "Laboratorio Di Statistica Con R" y está parcialmente documentado en italiano (utiliza nombres de funciones y argumentos en inglés).
```
```suggestion
Por ejemplo, el paquete [labstatR](https://cran.r-project.org/web/packages/labstatR/index.html) sirve de complemento al libro italiano "Laboratorio Di Statistica Con R" y está parcialmente documentado en italiano (utiliza nombres de funciones y argumentos en inglés).
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+Del mismo modo, [chilemapas](https://cran.r-project.org/web/packages/chilemapas/chilemapas.pdf) proporciona mapas simplificados para Chile, con documentación completa en español, incluidos los nombres de las funciones.
+rOpenSci [censo2017](https://docs.ropensci.org/censo2017/) por [Mauricio Vargas](/author/pachá-aka-mauricio-vargas-sepúlveda/) también está totalmente documentado en español.
+
+[^1]: La cita real es "Se puede utilizar un campo 'Idioma' para indicar si la documentación del paquete no está en inglés", lo que indica que el inglés sigue siendo el supuesto por defecto.
```
```suggestion
[^1]: La cita real es "Se puede utilizar el campo 'Language' para indicar si la documentación del paquete no está en inglés", lo que indica que el inglés sigue siendo el supuesto por defecto.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
El paquete [karel](https://cloud.r-project.org/web/packages/karel/index.html) tiene dos conjuntos de funciones, uno con nombres y documentación en inglés y otro en español.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Ambas son soluciones alternativas que llevan al límite lo que el lenguaje de programación R puede hacer en términos de funcionalidad multilingüe.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Por eso presentamos el proyecto de paquete experimental **rhelpi18n** [^2].
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
[^2]: El nombre proviene del [neurónimo](https://en.wikipedia.org/wiki/Numeronym) de "internacionalización".
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Este paquete es un banco de pruebas para implementar un soporte sin fallas para la documentación multilingüe con el objetivo de integrarlo en el propio R una vez que madure.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Además de presentar la documentación traducida, rhelpi18n también proporciona herramientas para que las personas que desarrollan creen nuevos módulos de traducción.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
                              package_path = glue_dir,  # Camino a la copia local del codigo fuente de glue
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
El proyecto está en sus primeras fases y las cosas avanzan rápido.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Parte del flujo de trabajo de las traducciones suele consistir en traducciones automáticas que posteriormente revisa una persona.
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Por ello, estamos considerando la posibilidad de añadir metadatos a las cadenas para señalar los casos de traducciones automáticas que no han sido revisadas por una persona o las traducciones que podrían estar ligeramente desfasadas (¡sería una pena no mostrar una traducción porque alguien arregló un pequeño error de tipeo en el texto original y aún no había sido revisada!)
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```suggestion
Si deseas probar el paquete e informar de cualquier problema, o si tienes ideas sobre cómo mejorarlo, puedes participar en el debate en la sección [_issues_ abiertos](https://github.com/eliocamp/rhelpi18n/issues) del paquete y uniéndote al [Grupo de trabajo de documentación multilingüe](https://github.com/RConsortium/multilingual-documentation-wg).
```

steffilazerte commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,228 @@
+  - documentation
+  - community
+# The summary below will be used by e.g. Mastodon preview cards
+description: "Multilingual documentation coming to an R package near you"
```
```suggestion
description: "Multilingual documentation coming to an R package near you!"
```

yabellini reviewed (commented):
@eliocamp here is my review of the Spanish translations. Let us know when is ready.  I will also ask you to add to the YAML the editors' data:

```
editor:
  - Steffi LaZerte
  - Yanina Bellini Saibene
```

steffilazerte commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,228 @@
+
+Although English is the de facto international language, there are [benefits](https://ropensci.org/multilingual-publishing/) to including resources in different languages.
+By reducing language barriers we end up with a larger, stronger, more creative community with more ideas and resources.
+This English hegemony is reflected in R by the use of English in its documentation, such as manual pages, function names, and argument names (why use `mean()` instead of `Mittelwert()`?)
```
```suggestion
The English hegemony is reflected in R by the use of English in its documentation, such as manual pages, function names, and argument names (why use `mean()` instead of `Mittelwert()`?)
```

steffilazerte commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.Rmd:
```suggestion
This is all in the early stages but things are moving fast.
```

steffilazerte commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,228 @@
+---
+slug: "2024-07-09-multilingual-documentation"
```
```suggestion
slug: "multilingual-documentation"
```

steffilazerte commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,217 @@
+---
+slug: 2024-07-09-multilingual-documentation
```
```suggestion
slug: multilingual-documentation
```

steffilazerte reviewed (commented):
Hi @eliocamp, thank you for your thoughtful consideration of the comments!
The article looks great and I really like the explanation you added to to the idea of native vs. package features.
I have a couple of super minor suggestions, including if you could fix the slug (no date in the slug, sorry I missed that in the first round; just remember to re-render the file). But other than that I'm good to go on my end.

@yabellini, will you handle the publication/social media, or should I?

yabellini commented:
@steffilazerte I can handle social media.  Thanks so much!!

steffilazerte commented:
And you'll merge tomorrow? (Just want to be sure 😅)

yabellini commented:
> And you'll merge tomorrow? (Just want to be sure 😅)

Yes! 😁

eliocamp commented:
Done!

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.Rmd:
```diff
@@ -0,0 +1,227 @@
+# Delete the package_version line below if your post is not about a package
+author:
+  - Elio Campitelli
+# Set the date below to the publication date of your post
```
```suggestion
editor:
  - Steffi LaZerte
  - Yanina Bellini Saibene
# Set the date below to the publication date of your post
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.Rmd:
```diff
@@ -0,0 +1,216 @@
+title: Documentación multilingüe en paquetes de R
+author:
+- Elio Campitelli
+date: '2024-07-09'
```
```suggestion
editor:
  - Steffi LaZerte
  - Yanina Bellini Saibene
date: '2024-07-09'
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.es.md:
```diff
@@ -0,0 +1,244 @@
+title: Documentación multilingüe en paquetes de R
+author:
+- Elio Campitelli
+date: '2024-07-09'
```
```suggestion
editor:
  - Steffi LaZerte
  - Yanina Bellini Saibene
date: '2024-07-09'
```

yabellini commented on content/blog/2024-07-09-multilingual-documentation-in-r/index.md:
```diff
@@ -0,0 +1,255 @@
+# Delete the package_version line below if your post is not about a package
+author:
+  - Elio Campitelli
+# Set the date below to the publication date of your post
```
```suggestion
editor:
  - Steffi LaZerte
  - Yanina Bellini Saibene
# Set the date below to the publication date of your post
```
